GAMEFLOW.md

```diff
@@ -0,0 +1,1306 @@
+    <th>Name</th>
+  </tr>
+  <tr>
+    <td>10</td>
+    <td>Crocodile</td>
+  </tr>
+  <tr>
+    <td>11</td>
+    <td>Alligator</td>
+  </tr>
+  <tr>
+    <td>16</td>
+    <td>Rat</td>
+  </tr>
+  <tr>
+    <td>17</td>
+    <td>Vole</td>
+  </tr>
+  <tr>
+    <td>20</td>
+    <td>Flying mutant</td>
+  </tr>
+  <tr>
+    <td>21</td>
+    <td>Grounded mutant (shooter)</td>
+  </tr>
+  <tr>
+    <td>22</td>
+    <td>Grounded mutant (non-shooter)</td>
+  </tr>
+</table>
+
+### Item validity
+The following object types are capable of being carried and dropped. The
+gameflow will ignore anything that is not in this list, and a suitable warning
+message will be produced in the log file.
+
+<table>
+  <tr valign="top" align="left">
+    <th>Object ID</th>
+    <th>Name</th>
+  </tr>
+  <tr>
+    <td>84</td>
+    <td>Pistols</td>
+  </tr>
+  <tr>
+    <td>85</td>
+    <td>Shotgun</td>
+  </tr>
+  <tr>
+    <td>86</td>
+    <td>Magnums</td>
+  </tr>
+  <tr>
+    <td>87</td>
+    <td>Uzis</td>
+  </tr>
+  <tr>
+    <td>89</td>
+    <td>Shotgun ammo</td>
+  </tr>
+  <tr>
+    <td>90</td>
+    <td>Magnum ammo</td>
+  </tr>
+  <tr>
+    <td>91</td>
+    <td>Uzi ammo</td>
+  </tr>
+  <tr>
+    <td>93</td>
+    <td>Small medipack</td>
+  </tr>
+  <tr>
+    <td>94</td>
+    <td>Large medipack</td>
+  </tr>
+  <tr>
+    <td>110</td>
+    <td>Puzzle1</td>
+  </tr>
+  <tr>
+    <td>111</td>
+    <td>Puzzle2</td>
+  </tr>
+  <tr>
+    <td>112</td>
+    <td>Puzzle3</td>
+  </tr>
+  <tr>
+    <td>113</td>
+    <td>Puzzle4</td>
+  </tr>
+  <tr>
+    <td>126</td>
+    <td>Lead bar</td>
+  </tr>
+  <tr>
+    <td>129</td>
+    <td>Key1</td>
+  </tr>
+  <tr>
+    <td>130</td>
+    <td>Key2</td>
+  </tr>
+  <tr>
+    <td>131</td>
+    <td>Key3</td>
+  </tr>
+  <tr>
+    <td>132</td>
+    <td>Key4</td>
+  </tr>
+  <tr>
+    <td>141</td>
+    <td>Pickup1</td>
+  </tr>
+  <tr>
+    <td>142</td>
+    <td>Pickup2</td>
+  </tr>
+  <tr>
+    <td>144</td>
+    <td>Scion (à la Pierre)</td>
+  </tr>
+</table>
+
+## Injections
+Injections defined in the global gameflow will by default be applied to each
+level. Individual levels can also specify their own specific injections to
+include. In that case, the global injections are merged with the level's when
+the level loads.
+
+Individual levels can set `inherit_injections` to `false` and as a result they
+will not use the global injection files. If those levels have their own
+injections defined, only those will be used. And of course, if they have none
+defined, nothing will be injected.
+
+_Disclaimer: Custom levels should not use the injections mechanism and instead
+should provide data that is already correct and consistent. Reports of bugs
+about injection files not working for custom levels will not be considered. Note
+however that the injections that relate to Lara can work in custom levels
+that do not modify Lara's default mesh structure or animations. These injection
+files are based on the original Lara model._
```

Comment:
I agree with this disclaimer mostly but aren't injections required if builders want animations like the jump twist? This makes it seem like they should never be used. Can the animations like jump twist just be added to Lara's animations normally? Is there a required animation slot? Same for underwater roll, responsive jumping, etc.

Reply:
Yes, so I'm hoping the extra note there covers it, so builders know that provided they have used an original Lara model, the animation injections will work fine. They could create the animations manually, and all the state changes, commands etc - I'm not sure if we want to get into supporting achieving that though as it's quite intricate. And yes, you're right, it needs the new anims in specific slots.